kymatio/frontend/base_frontend.py

```diff
@@ -13,6 +14,17 @@ def build(self):
         self.filters = self.create_filters() """
         raise NotImplementedError
 
+    def _check_filterbanks(psi1s, psi2s):
+        # Check that the Nyquist frequency of each filter, 2**(-j), is at least
+        # one octave above its frequency xi
+        assert all((psi1["xi"] < 0.5/(2**psi1["j"])) for psi1 in psi1s)
+
+        # Check that (j1 < j2) implies (xi1 > xi2)
+        psi_generator = itertools.product(psi1s, psi2s)
+        condition = lambda psi1_or_2: psi1_or_2[0]['j'] < psi1_or_2[1]['j']
+        implication = lambda psi1_or_2: psi1_or_2[0]['xi'] > psi1_or_2[1]['xi']
+        assert all(map(implication, filter(condition, psi_generator)))
```

Comment:
Briefly explain what this line does?

Reply:
6c316cb6aea38d6de736d04ac9e2de81f943c86e